include/lsst/afw/table/Schema.h

```diff
@@ -341,10 +354,52 @@ class SubSchema {
     typedef detail::SchemaImpl Impl;
 public:
 
+    //@{
+    /// Join strings using the field delimiter appropriate for this Schema
+    std::string join(std::string const & a, std::string const & b) const;
+    std::string join(std::string const & a, std::string const & b, std::string const & c) const {
+        return join(join(a, b), c);
+    }
+    std::string join(
+        std::string const & a, std::string const & b, std::string const & c, std::string const & d
+    ) const {
+        return join(join(a, b), join(c, d));
+    }
+    //@}
+
     /// @brief Find a nested SchemaItem by name.
     template <typename T>
     SchemaItem<T> find(std::string const & name) const;
 
+    /**
+     *  @brief Find a nested SchemaItem by name and run a functor on it.
+     *
+     *  Names corresponding to named subfields are not accepted.
+     *  The given functor must have an overloaded function call
+     *  operator that accepts any SchemaItem type (the same as
+     *  a functor provided to apply or Schema::forEach).
+     */
+    template <typename F>
+    void find(std::string const & name, F && func) const {
```

Comment:
This function needs to have a templated `operator()`, which `std::function` doesn't support.